homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +184,187 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+        self._brightness = None
+        self._hs_color = None
+        self._white_value = None
+        self._current_effect = None
+        self._bulb = WifiLedBulb(device[CONF_HOST])
+        self._last_brightness = 255
+        self._last_hs_color = color_util.color_RGB_to_hs(255, 255, 255)
+        self._ip_address = device[CONF_HOST]
+        self._effect_speed = effect_speed
+        self._mode = None
+        self._get_rgbw = None
+        self._get_rgb = None
+
+    async def async_remove_light(self, device: dict):
+        """Remove a bulb device when it is removed from options."""
+
+        bulb_id = device["device_id"]
+
+        if self._unique_id != bulb_id:
+            return
+
+        entity_registry = await self.hass.helpers.entity_registry.async_get_registry()
+        entity_entry = entity_registry.async_get(self.entity_id)
+
+        device_registry = await self.hass.helpers.device_registry.async_get_registry()
+        device_entry = device_registry.async_get(entity_entry.device_id)
+
+        if (
+            len(
+                async_entries_for_device(
+                    entity_registry,
+                    entity_entry.device_id,
+                    include_disabled_entities=True,
+                )
+            )
+            == 1
+        ):
+            # If only this entity exists on this device, remove the device.
+            device_registry.async_remove_device(device_entry.id)
+
+        entity_registry.async_remove(self.entity_id)
+
+    async def async_added_to_hass(self):
+        """Run when the entity is about to be added to hass."""
+        await super().async_added_to_hass()
+
+        async_dispatcher_connect(
+            self.hass, SIGNAL_REMOVE_DEVICE, self.async_remove_light
+        )
+
+    def update_bulb_info(self):
+        """Update the bulb information."""
+        self._bulb.update_state()
+        self._get_rgbw = self._bulb.getRgbw()
+        self._get_rgb = self._bulb.getRgb()
+
+    @Throttle(timedelta(seconds=DEFAULT_SCAN_INTERVAL))
+    async def async_update(self):
+        """Fetch the data from this light bulb."""
+        if self._bulb_updater:
+            await self._bulb_updater.async_update()
+
+        await self.hass.async_add_executor_job(self.update_bulb_info)
+
+        if self._bulb.mode == "ww":
+            self._mode = MODE_WHITE
+        elif self._bulb.rgbwcapable:
+            self._mode = MODE_RGBW
+        else:
+            self._mode = MODE_RGB
+
+        self._white_value = self._get_rgbw[3]
+
+        if self._mode == MODE_WHITE:
+            self._brightness = self._white_value
+        else:
+            self._brightness = self._bulb.brightness
+
+        self._hs_color = color_util.color_RGB_to_hs(*self._get_rgb)
+
+        self._current_effect = self._bulb.raw_state[3]
+
+        if self._bulb.is_on and self._brightness > 0:
+            self._state = True
+        else:
+            self._state = False
+
+        self.async_write_ha_state()
```

Comment:
The automatic entity update will already schedule a state update.
```suggestion
```

Reply:
OK removed.